src/Microsoft.OData.Core/ODataServiceColectionExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataServiceColectionExtensions.cs" company="Microsoft">
+// Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using Microsoft.OData;
+using Microsoft.OData.Edm;
+using Microsoft.OData.Json;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for <see cref="IServiceCollection"/>.
+    /// </summary>
+    public static class ODataServiceColectionExtensions
+    {
+        /// <summary>
+        /// Adds the default OData services to the <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
+        /// <returns>The <see cref="IServiceCollection"/> instance itself.</returns>
+        public static IServiceCollection AddDefaultODataServices(this IServiceCollection services)
+        {
+            return AddDefaultODataServices(services, ODataVersion.V4);
+        }
+
+        /// <summary>
+        /// Adds the default OData services to the <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
+        /// <param name="odataVersion">ODataVersion for the default services.</param>
+        /// <returns>The <see cref="IServiceCollection"/> instance itself</returns>
+        public static IServiceCollection AddDefaultODataServices(this IServiceCollection services, ODataVersion odataVersion)
+        {
+            if (services is null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            services.AddSingleton<IJsonReaderFactory, DefaultJsonReaderFactory>();
+            services.AddSingleton<IJsonWriterFactory, DefaultJsonWriterFactory>();
+            services.AddSingleton(sp => ODataMediaTypeResolver.GetMediaTypeResolver(null));
+            services.AddScoped<ODataMessageInfo>();
+            services.AddScoped(sp => new ODataMessageReaderSettings(odataVersion));
+            services.AddScoped(sp => new ODataMessageWriterSettings(odataVersion));

```

Comment:
Unless it's an extreme undertaking, you should collect COGS stats comparing the previous use of `ServicePrototype` to register `ODataMessageWriterSettings` and `ODataMessageReaderSettings` plus cloning vs doing it in this new way.

Reply:
Ran some benchmarks for the two methods and this is what I got: 
<img width="620" alt="xzz" src="https://github.com/OData/odata.net/assets/25525526/bbb78a5a-110c-4ded-a237-f84d04ef467e">

Reply:
Awesome! 75-80% perf improvement and 50% allocation reduction if I'm reading that right. Nicely done!